### PR TITLE
ES6 docs Part 4

### PR DESCRIPTION
**Pull request by BenAtEventbrite**
- rest parameters
- default parameters
- spread operator

### CODE REVIEW

File: es6/README.md [hunk 1]
```diff
@@ -459,13 +459,127 @@ For more on arrow functions, read [_Learning ES6: Arrow Functions_](http://www.e
...
 **[⬆ back to top](#table-of-contents)**
 
 ### Default Parameters
 
-Coming soon...
+Use default parameters in the function header instead of mutating parameters in the function body:
+
+```js
+// good
+const getData = (options, useCache=true) => {
```
Comment (kaylieEB) [inline es6/README.md]: isn't it preferred to have spaces between default params? (or at least I prefer it haha) http://eslint.org/docs/rules/space-infix-ops

File: es6/README.md [hunk 2]
```diff
@@ -459,13 +459,127 @@ For more on arrow functions, read [_Learning ES6: Arrow Functions_](http://www.e
...
+    let data;
+
+	if (useCache) {
+		// get data from cache
```
Comment (kaylieEB) [inline es6/README.md]: maybe add here `data = options.cache;` for clarity

File: es6/README.md [hunk 3]
```diff
@@ -487,6 +487,84 @@ For more on rest parameters, read [_Learning ES6: Rest & Spread Operators_](http
...
 ### Default Parameters
 
+Use default parameters in the function header instead of mutating parameters in the function body:
+
+```js
+// good
+const getData = (options, useCache=true) => {
+    let data;
+
+	if (useCache) {
```
Comment (vincent-eb) [inline es6/README.md]: indentation looks a little off on the `if` blocks in each of the examples

Comment (BenAtEventbrite) [inline es6/README.md]: weird! i'll fix!

File: es6/README.md (hunk 1 above)
Comment (BenAtEventbrite) [inline es6/README.md]: oOo. that must be a new rule. we don't have one, but yeah I agree it'd be nice

File: es6/README.md (hunk 2 above)
Comment (BenAtEventbrite) [inline es6/README.md]: Well the data wouldn't be in `options`. The cache would probably be some global object

**Pull request merged**